src/foam/dao/RemoveSink.java

```diff
@@ -0,0 +1,30 @@
+package foam.dao;
+
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.core.X;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+
+public class RemoveSink {
+
+  public RemoveSink(final DAO dao) {
```

Comment:
RemoveSink's constructor should just store the 'dao' and do nothing else. It should have a put() method that does dao.remove(x, obj);